src/Swift.Bindings/tests/DemanglerTests/BasicDemanglingTests.cs

```diff
@@ -0,0 +1,71 @@
+using BindingsGeneration.Demangling;
+using Xunit;
+
+namespace BindingsGeneration.Tests;
+
+public class BasicDemanglingTests : IClassFixture<BasicDemanglingTests.TestFixture>
+{
+    private readonly TestFixture _fixture;
+
+    public BasicDemanglingTests(TestFixture fixture)
+    {
+        _fixture = fixture;
+    }
+
+    public class TestFixture
+    {
+        static TestFixture()
+        {
+        }
+
+        private static void InitializeResources()
+        {
+        }
+    }
+
+    [Fact]
+    public void TestProtocolWitnessTable()
+    {
+        var symbol = "_$s20GenericTestFramework6ThingyCAA7StanleyAAWP";
+        var demangler = new Swift5Demangler (symbol);
+        var result = demangler.Run ();
+        var protoWitnessReduction = result as ProtocolWitnessTableReduction;
+        Assert.NotNull(protoWitnessReduction);
+        Assert.Equal("GenericTestFramework.Thingy", protoWitnessReduction.ImplementingType.Name);
+        Assert.Equal("GenericTestFramework.Stanley", protoWitnessReduction.ProtocolType.Name);
+    }
+
+    [Fact]
+    public void TestOtherProtocolWitnessTable()
+    {
+        var symbol = "_$s20GenericTestFramework6ThingyCAA8IsItRealAAWP";
+        var demangler = new Swift5Demangler (symbol);
+        var result = demangler.Run ();
+        var protoWitnessReduction = result as ProtocolWitnessTableReduction;
+        Assert.NotNull(protoWitnessReduction);
+        Assert.Equal("GenericTestFramework.Thingy", protoWitnessReduction.ImplementingType.Name);
+        Assert.Equal("GenericTestFramework.IsItReal", protoWitnessReduction.ProtocolType.Name);
+    }
+
+    [Fact]
+    public void TestFailDemangleNonsense()
+    {
+        var symbol = "_$ThisIsJustGarbage";
+        var demangler = new Swift5Demangler (symbol);
+        var result = demangler.Run ();
+        var err = result as ReductionError;
```

Comment:
What is expected to happen if we would instead do `result as ProtocolWitnessTableReduction`? I

Reply:
It would absolutely return null. The test is there to ensure that a symbol that is obviously not a Swift mangled symbol will correctly generate an error. Anything other cast will fail.